pandas/core/indexing.py

```diff
@@ -1670,6 +1671,12 @@ def _setitem_with_indexer_split_path(self, indexer, value):
                 # We are trying to set N values into M entries of a single
                 #  column, which is invalid for N != M
                 # Exclude zero-len for e.g. boolean masking that is all-false
+
+                if len(value) == 1 and not is_integer(info_axis):
+                    # This is a case like df.iloc[:3, [1]] = [0]
+                    #  where we treat as df.iloc[:3, 1] = 0
+                    return self._setitem_with_indexer((pi, info_axis[0]), value[0])
```

Comment:
Is there a reason for not using ``_setitem_single_column``? Seems to work when trying out locally. LGTM otherwise

Reply:
either way works.  this seemed like a more obvious "we are reducing this to a simpler case" representation

Reply:
Ok, this is fine with me then.